.github/workflows/codeql.yml

```diff
@@ -0,0 +1,51 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  workflow_dispatch:
```

Comment:
is this needed? 🤔

Reply:
It is a manual trigger. It allows maintainers to go to the Actions tab and to click the button "Start Workflow". Not mandatory, but quite useful.

Reply:
Ok sure, let's keep it 👍